pkg/skaffold/build/cache/hash_test.go

```diff
@@ -39,46 +40,43 @@ var mockCacheHasher = func(s string) (string, error) {
 	return s, nil
 }
 
-var fakeArtifactConfig = func(a *latest.Artifact, devMode bool) (string, error) {
+var fakeArtifactConfig = func(a *latest.Artifact, mode config.RunMode) (string, error) {
 	if a.ArtifactType.DockerArtifact != nil {
 		return "docker/target=" + a.ArtifactType.DockerArtifact.Target, nil
 	}
-	if devMode {
-		return "devmode", nil
-	}
-	return "other", nil
+	return string(mode), nil
 }
 
 func TestGetHashForArtifact(t *testing.T) {
 	tests := []struct {
 		description  string
 		dependencies []string
 		artifact     *latest.Artifact
-		devMode      bool
+		mode         config.RunMode
 		expected     string
 	}{
 		{
 			description:  "hash for artifact",
 			dependencies: []string{"a", "b"},
 			artifact:     &latest.Artifact{},
-			expected:     "1caa15f7ce87536bddbac30a39768e8e3b212bf591f9b64926fa50c40b614c66",
+			expected:     "d99ab295a682897269b4db0fe7c136ea1ecd542150fa224ee03155b4e3e995d9",
```

Comment:
Can you explain why the hash for the artifact changed?

Reply:
That's because I modified the hash calculation function:
<img width="544" alt="image" src="https://user-images.githubusercontent.com/39389231/89854961-e4f2cb00-dbb2-11ea-837d-007e7ed1486f.png">

Reply:
Thanks. This would mean, if i run `skaffold dev` and then `skaffold debug` the artifact hash will change and a skaffold will the image again. 

This is not optimal.

Reply:
you're right. I've messed this part up a bit. I'll close the PR temporarily until I rethink this.

Reply:
This is what we want *if* the options are different (e.g., the Go GCFLAGS disable optimizations etc that we would want to keep for `build`).  But as is in this PR, we will do this for all images.

I'm thinking this may need to be tied into #3977.

Reply:
now I directly use the evaluated build args for the hash. This includes evaluated env for the buildpacks builder which already has the `GOOGLE_DEVMODE=1` so we hash different between dev and other modes and don't need this hack:
![image](https://user-images.githubusercontent.com/39389231/90038336-ecb18d00-dce2-11ea-88ad-19f92331e374.png)